### ISSUE TITLE
Unique lyrics?

### ISSUE THREAD

**Issue by jws325** [robbiebarrat/rapping-neural-network#2]
This script works well for reordering existing bars.

Does anyone know a way to generate unique lyrics?
Maybe by taking multiple lines and combining them, or using paraphrases?

**Comment by robbiebarrat** [comment]
You could probably try:
1. another neural network to combine lines (this would work the best but it would be a bit hard).
2. a markov chain (this would be easiest in my opinion).
3. think like, 'mad libs' with various parts of speech or key words to generate lines (I have no idea if this would work well).

**Comment by mikeres0** [comment]
If you were going to use a markov chain this could be helpful https://github.com/jsvine/markovify

**Comment by robbiebarrat** [comment]
Oh wow - https://github.com/jsvine/markovify could be really helpful. I'll experiment with it this afternoon and if I get any good results I'll incorporate it.

**Comment by mikeres0** [comment]
@robbiebarrat let me know how it goes... Currently producing rap lyrics for the office 💯

**Comment by ktxxt** [comment]
I'm also excited towards unique lyrics. Let us know how it goes :) :100:

**Comment by ufaceni** [comment]
@robbiebarrat, have you tried to do anything with Markov Chain?

**Comment by robbiebarrat** [comment]
i'm super busy with college applications and some finals for courses i'm taking at a nearby university right now, i've been messing around with markov chains and other stuff in my spare time though. 

I'm trying to hook up NLTK (the python library), so it will be able to make more coherent sentences by only replacing nouns, not having multiple verbs in a row, and just following basic rules of writing sentences. If the NLTK stuff is successful, I could probably just make another neural network in charge of writing sentences (different parts of speech could be assigned different numbers, it would actually be very easy to format this for a neural net), and i feel like having a neural network doing this would be more true to my vision for this project, as opposed to a markov chain.

Ideally, I'll make a huge update incorporating this over the weekend sometime (maybe Friday night). I'll also update you guys here with some sample songs/lyrics the updated program writes.
I'm thinking about making it a separate python file, (separate from deeprap.py at least) - just so that if you still wanted to use the old program that just rips lines directly from songs you could. If i do this, i'll probably make a 'wrapper' program too - one that you can just run and it will do everything by calling the other programs.

If you don't notice anything go up by Saturday, feel free to post something here and remind me - my memory isn't too great

**Comment by ktxxt** [comment]
@robbiebarrat a reminder :P saturdays over. my memory isn't great neither :D

**Comment by robbiebarrat** [comment]
Oh my goodness - that was 20 days ago. Yikes - I'm really sorry about that. 

I just did a huge update to deeprap.py where I took care of a bunch of major issues, such as the part about "manually selecting good lines", and I improved functionality/cleaned up the code a bunch. I'll be working on having it write individual lines throughout the week, as I'm almost done applying to colleges (I have like, two or three left but their applications aren't due until around the 20th, so I have some time to do this first!)

tl;dr: Thanks for being patient, I just improved deeprap.py greatly, I'll be working on the individual line writing this week.

**Comment by robbiebarrat** [comment]
@ufaceni @ktxxt @mikeres0 
I just updated it - now it can write songs word by word, so yes it can come up with unique lyrics.
Sorry it took so long! Hope everyone enjoys it!
